systemtest/src/main/java/io/strimzi/systemtest/resources/crd/KafkaBridgeResource.java

```diff
@@ -5,124 +5,51 @@
     }
-
-    private static KafkaBridgeBuilder defaultKafkaBridge(KafkaBridge kafkaBridge, String name, String kafkaClusterName, String bootstrap, int kafkaBridgeReplicas) {
-        return new KafkaBridgeBuilder(kafkaBridge)
-            .withNewMetadata()
-                .withName(name)
-                .withNamespace(ResourceManager.kubeClient().getNamespace())
-                .withClusterName(kafkaClusterName)
-            .endMetadata()
-            .editSpec()
-                .withBootstrapServers(bootstrap)
-                .withReplicas(kafkaBridgeReplicas)
-                .withNewInlineLogging()
-                    .addToLoggers("bridge.root.logger", Environment.STRIMZI_COMPONENTS_LOG_LEVEL)
-                .endInlineLogging()
-            .endSpec();
+    @Override
+    public void create(KafkaBridge resource) {
+        kafkaBridgeClient().inNamespace(resource.getMetadata().getNamespace()).createOrReplace(resource);
     }
-
-    public static KafkaBridge createAndWaitForReadiness(KafkaBridge kafkaBridge) {
-        KubernetesResource.deployNetworkPolicyForResource(kafkaBridge, KafkaBridgeResources.deploymentName(kafkaBridge.getMetadata().getName()));
-
-        TestUtils.waitFor("KafkaBridge creation", Constants.POLL_INTERVAL_FOR_RESOURCE_CREATION, CR_CREATION_TIMEOUT,
-            () -> {
-                try {
-                    kafkaBridgeClient().inNamespace(ResourceManager.kubeClient().getNamespace()).createOrReplace(kafkaBridge);
-                    return true;
-                } catch (KubernetesClientException e) {
-                    if (e.getMessage().contains("object is being deleted")) {
-                        return false;
-                    } else {
-                        throw e;
-                    }
-                }
-            }
-        );
-        return waitFor(deleteLater(kafkaBridge));
+    @Override
+    public void delete(KafkaBridge resource) throws Exception {
+        kafkaBridgeClient().inNamespace(resource.getMetadata().getNamespace()).withName(
+            resource.getMetadata().getName()).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete();
     }
-
-    public static KafkaBridge kafkaBridgeWithoutWait(KafkaBridge kafkaBridge) {
-        kafkaBridgeClient().inNamespace(ResourceManager.kubeClient().getNamespace()).createOrReplace(kafkaBridge);
-        return kafkaBridge;
+    @Override
+    public boolean isReady(KafkaBridge resource) {
+        return ResourceManager.waitForResourceStatus(kafkaBridgeClient(), resource, CustomResourceStatus.Ready);
     }
-
-    public static void deleteKafkaBridgeWithoutWait(String resourceName) {
-        kafkaBridgeClient().inNamespace(ResourceManager.kubeClient().getNamespace()).withName(resourceName).withPropagationPolicy(DeletionPropagation.FOREGROUND).delete();
+    @Override
+    public void replaceResource(KafkaBridge existing, KafkaBridge newResource) {
```

Comment:
How does this work? I can see from the JavaDoc that it should replace a resource in Kubernetes. But I do not see it done here. It seems to only update one other resource without the Kube API being involved.

I also did not find any use of this method.

Reply:
```
synchronized (this) {
                    T updated = type.get(resource.getMetadata().getNamespace(), resource.getMetadata().getName());
                    type.replaceResource(resource, updated);
                }
```
here :)

Reply:
Link to the file would be helpful to understand what the whole code is doing.